### DISCUSSION TITLE
Detecting right or left click on a block

### DISCUSSION THREAD

**Discussion by Mqzn** [Minestom/Minestom#1790] [Help] [answered]
I can't seem to detect when  I right or left click a block
when i left click it detects that the hand is MAIN , otherwise it doesn't consider right clicking an interaction 

Here's my code :
![image](https://user-images.githubusercontent.com/72102072/225747884-db94a0d6-1d0a-4b95-85db-9ddd5c1b783f.png)

**Comment by emortaldev** [comment]
Hand animation does not play for right click and so won't trigger that event. You probably want `PlayerBlockInteractEvent` or `PlayerUseItemOnBlockEvent` or `PlayerUseItemEvent`.

**Reply by Mqzn** [reply]
After several debugging , I found out that `PlayerUseItemOnBlockEvent` and `PlayerBlockInteractEvent` both detects Right clicking **only**

Still couldn't find a way to detect left clicking on a block. Moreover, I also tried using `BlockBreakEvent` but didn't work
![image](https://user-images.githubusercontent.com/72102072/226025850-07a5c588-2eb9-4452-8aa6-7c3cd1463313.png)

**Reply by Mqzn** [reply]
I have found out that `PlayerHandAnimationEvent` detects left clicking, but there's a small problem,
![image](https://user-images.githubusercontent.com/72102072/226030450-3181661e-7534-46e5-9cb7-8b65da766494.png)
the execution occurs twice 

this is the current code
![image](https://user-images.githubusercontent.com/72102072/226030970-9690739c-e964-4ad7-8f01-7231e94bb088.png)

**Reply by emortaldev** [reply]
`PlayerStartDiggingEvent` should do the trick as said by @LooFifteen

**Comment by LooFifteen** [comment, marked as answer]
`PlayerBlockInteractionEvent` can be used for right clicking when checking which hand is doing it. `PlayerStartDiggingEvent` can be used for left clicking.
